packages/api/lib/writeRecords/write-granules.js

```diff
@@ -462,34 +491,59 @@ const _writeGranuleRecords = async (params) => {
     writeConstraints = true,
   } = params;
   let pgGranule;
+  let writePgGranuleResult;
 
   log.info('About to write granule record %j to PostgreSQL', postgresGranuleRecord);
   log.info('About to write granule record %j to DynamoDB', apiGranuleRecord);
 
   try {
     await createRejectableTransaction(knex, async (trx) => {
-      pgGranule = await _writePostgresGranuleViaTransaction({
+      writePgGranuleResult = await _writePostgresGranuleViaTransaction({
         granuleRecord: postgresGranuleRecord,
         executionCumulusId,
         trx,
         granulePgModel,
         writeConstraints,
       });
-
-      // Future: refactor to cover the entire object?
-      // Ensure PG default createdAt value is propagated to DynamoDB/ES
-      // in the case where _writeGranule is called without createdAt set
-      if (!apiGranuleRecord.createdAt) {
-        apiGranuleRecord.createdAt = pgGranule.created_at.getTime();
+      if (writePgGranuleResult.status === 'dropped') {
+        return;
       }
+      pgGranule = writePgGranuleResult.pgGranule;
 
-      await granuleModel.storeGranule(apiGranuleRecord, writeConstraints);
-      await upsertGranule({
-        esClient,
-        updates: apiGranuleRecord,
-        index: process.env.ES_INDEX,
-      }, writeConstraints);
+      if (writeConstraints && isStatusActiveState(pgGranule.status)) {
+        // pgGranule was updated, but with writeConstraints conditions, so only some values were
+        // updated. we need to ensure the correct values are propogated to Dynamo and ES
+        const limitedUpdateApiGranuleRecord = await translatePostgresGranuleToApiGranule({
```

Comment:
I commented here before - I'm still a bit wary of this, because it only works *if* the write limits stay current.    Meaning, so long as createdAt, updatedAt, status and timestamp are the only modified fields, this works, but if the upsert conditionals change to allow files, this could break.     I think we need to ensure that can't happen for maintainability reasons, either via test, annotation or refactor here.

Reply:
I'm not sure how this can be done via refactor.
I _think_ the new tests I've written do test that the only values changed are the ones allowed, as well as the one test that was unexpectedly fixed. If you disagree, what more do you think should be tested?
I assume by annotation you mean by comments. I will add some additional comments, any suggested edits or additions to those comments would be very welcome.